plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java

```diff
@@ -236,5 +245,80 @@ public void testFilesTable()
                         "('split_offsets', 'array(bigint)', '', '')," +
                         "('equality_ids', 'array(integer)', '', '')");
         assertQuerySucceeds("SELECT * FROM test_schema.\"test_table$files\"");
+
+        MaterializedResult result = computeActual("SELECT file_format, record_count, lower_bounds, upper_bounds FROM test_schema.\"test_table$files\" order by file_path");
+
+        MaterializedResult expectedStatistics =
+                MaterializedResult.resultBuilder(getSession(), VARCHAR, BIGINT, new MapType(INTEGER, VARCHAR, new TypeOperators()),
+                                new MapType(IntegerType.INTEGER, VARCHAR, new TypeOperators()))
+                        .row("ORC", 1L, Map.of(1, "0", 2, "2019-09-08"), Map.of(1, "0", 2, "2019-09-08"))
+                        .row("ORC", 2L, Map.of(1, "1", 2, "2019-09-09"), Map.of(1, "2", 2, "2019-09-09"))
+                        .row("ORC", 1L, Map.of(1, "3", 2, "2019-09-09"), Map.of(1, "3", 2, "2019-09-09"))
+                        .row("ORC", 2L, Map.of(1, "4", 2, "2019-09-10"), Map.of(1, "5", 2, "2019-09-10"))
+                        .build();
+
+        assertEquals(result, expectedStatistics);
+    }
+
+    @Test
+    public void testAllFilesTable()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "all_files_table", "(a int, b VARCHAR)", ImmutableList.of("(1, 'a')"))) {
+            assertQuery("SHOW COLUMNS FROM \"" + table.getName() + "$all_files\"",
+                    "VALUES ('content', 'integer', '', '')," +
+                            "('file_path', 'varchar', '', '')," +
+                            "('file_format', 'varchar', '', '')," +
+                            "('record_count', 'bigint', '', '')," +
+                            "('file_size_in_bytes', 'bigint', '', '')," +
+                            "('column_sizes', 'map(integer, bigint)', '', '')," +
+                            "('value_counts', 'map(integer, bigint)', '', '')," +
+                            "('null_value_counts', 'map(integer, bigint)', '', '')," +
+                            "('nan_value_counts', 'map(integer, bigint)', '', '')," +
+                            "('lower_bounds', 'map(integer, varchar)', '', '')," +
+                            "('upper_bounds', 'map(integer, varchar)', '', '')," +
+                            "('key_metadata', 'varbinary', '', '')," +
+                            "('split_offsets', 'array(bigint)', '', '')," +
+                            "('equality_ids', 'array(integer)', '', '')");
+
+            assertQuerySucceeds("SELECT * FROM \"" + table.getName() + "$all_files\"");
```

Comment:
Is it possible to verify all fields' values? Currently, this assertion confirms only query success and some fields (e.g. `key_metadata`, `split_offsets` & `equality_ids`) are null.

Reply:
@ebyhr I'm testing some of the important fields like `file_format, record_count, lower_bounds, upper_bounds` in the query/section below. (like 286)
Adding these fields were also suggested by @findinpath 

Do you want me to remove the above query as it is just validating whether the query is successful or not?

Reply:
Agree with adding those properties, but I hesitate to approve the PR without test (having values).

Reply:
added all the columns

Reply:
I have updated the `testFilesTable` as well, earlier `testFilesTable` didn't have any validation.